src/figuresClasses.ts

```diff
@@ -1,19 +1,76 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 export interface Figure {
-
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0
+      || b <= 0
+      || c <= 0
+      || a + b <= c
+      || a + c <= b
+      || c + b <= a) {
+      throw new Error('Please enter the correct value of the sides');
+    }
+  }
+
+  getArea(): number {
+    const halfP: number = (this.a + this.b + this.c) / 2;
 
+    return Number(Math.sqrt(
+      halfP * (halfP - this.a) * (halfP - this.b) * (halfP - this.c),
+    ).toFixed(2));
```

Comment:
Why do you need toFixed if you know how to do it with * 100 / 100
Remove it rounding to the separate function & reuse in all cases
also, split it with variable
```suggestion
    return getRounding(square);
```